src/ledger/ledger-wrapper.ts

```diff
@@ -76,17 +78,25 @@ export const encodeDerivationPath = (derivationPath: string) => {
   return `${dataLength}${data}`
 }
 
+const LedgerSubjects = () => ({
+  onProgressSubject: new Subject<string | undefined>(),
+})
+
 export type LedgerOptions = {
   transport: typeof TransportWebHID
+  ledgerSubjects?: ReturnType<typeof LedgerSubjects>
 }
 
-export const LedgerWrapper = ({ transport }: LedgerOptions) => {
+export const LedgerWrapper = ({ transport, ledgerSubjects }: LedgerOptions) => {
+  const { onProgressSubject } = ledgerSubjects || LedgerSubjects()
```

Comment:
had an issue with subjects when deconstructing them. If you're providing subjects it might not behave as expected